pydantic/main.py

```diff
@@ -214,6 +214,8 @@ class BaseModel(_repr.Representation, metaclass=ModelMetaclass):
         __pydantic_generic_metadata__: typing.ClassVar[_generics.PydanticGenericMetadata]
         __pydantic_parent_namespace__: typing.ClassVar[dict[str, Any] | None]
     else:
+        model_fields = {}
+        __pydantic_decorators__ = _decorators.DecoratorInfos()
```

Comment:
A comment here explaining might be helpful? If I didn't know this was part of the PR I wouldn't think twice about this, but I cannot imagine how this fixes the problem (or why `model_fields` was being accessed in the first place).

Reply:
We could use getattr there instead, but it seems to me better to have the type annotation be correct than to lie twice. But if it's possible for the class attributes to cause issues then we can keep them "missing" and just getattr in the `_generate_schema.model_schema`